OSDOCS#12905: Resolving content from multiple catalogs (#88394)
openshift/openshift-docs: extensions/catalogs/catalog-content-resolution.adoc

:_mod-docs-content-type: ASSEMBLY
[id="catalog-content-resolution"]
= Catalog content resolution
include::_attributes/common-attributes.adoc[]
:context: catalog-content-resolution

toc::[]

When you specify the cluster extension you want to install in a custom resource (CR), {olmv1-first} uses catalog selection to resolve what content is installed.

You can perform the following actions to control the selection of catalog content:

> * Specify metadata to select the catalog by name or labels

Reviewer: Technically, we don't support name directly. We create a label with the name that can then be selected upon

Reviewer 2: @tmshort The assumption is that the name label would be available by-default with cluster extensions in a catalog?

Author: Addressed: https://github.com/openshift/openshift-docs/pull/88394/commits/341b48ef8fe228c72f1fd1fd70bf11bcc76699d3

> * Specify labels to select the catalog
> * Use match expressions to perform complex filtering across catalogs
> * Set catalog priority

Reviewer 3: These are full sentences:
```suggestion
* Specify labels to select the catalog.
* Use match expressions to perform complex filtering across catalogs.
* Set catalog priority.
```

> If you do not specify any catalog selection criteria, {olmv1-first} might select from any available catalog on the cluster that provides the requested package.

Reviewer 3: Might? That suggests it does or doesn't select the extension. Is that right?

```suggestion
If you do not specify any catalog selection criteria, {olmv1-first} selects the extension from any available catalog on the cluster that provides the requested package.
```

> By default, bundles that are not deprecated are preferred over deprecated bundles during resolution.

Reviewer 3: I think this reads a bit better. 
```suggestion
By default, during resolution bundles that are not deprecated are preferred over deprecated bundles.
```
